test/test-runtime-onMessage.js

```diff
@@ -0,0 +1,108 @@
+"use strict";
+
+const {assert} = require("chai");
+const sinon = require("sinon");
+
+const {setupTestDOMWindow} = require("./setup");
+
+describe("browser-polyfill", () => {
+  describe("wrapped runtime.onMessage listener", () => {
+    it("keep track of the listeners added", () => {
+      const messageListener = sinon.spy();
+
+      const fakeChrome = {
+        runtime: {
+          lastError: undefined,
+          onMessage: {
+            addListener: sinon.spy(),
+            hasListener: sinon.stub(),
+            removeListener: sinon.spy(),
+          },
+        },
+      };
+
+      return setupTestDOMWindow(fakeChrome).then(window => {
+        fakeChrome.runtime.onMessage.hasListener
+          .onFirstCall().returns(false)
+          .onSecondCall().returns(true);
```

Comment:
hehehe, sorry :-)

I agree, in this case implementing the "has/add/remove" logic is pretty simple and it makes the test much more readable.

I've applied this change in last pushed updates (in bed17c0), which now wraps with sinon spies a simple "has/add/remove" implementation.